cpg_workflows/jobs/picard.py

```diff
@@ -132,7 +132,16 @@ def markdup(
     resource = HIGHMEM.request_resources(ncpu=4)
     # enough for input BAM and output CRAM
     resource.attach_disk_storage_gb = 250
-    resource.set_to_job(j)
+
+    # check for a memory override for impossible sequencing groups
+    if (memory_override := get_config()['resource_overrides'].get('picard')) is not None:
+        assert isinstance(memory_override, int)
+        # Hail will select the right number of CPUs based on RAM request
+        j.memory(f'{memory_override}G')
+        j.storage(f'{resource.get_storage_gb()}G')
+    else:
+        resource.set_to_job(j)
```

Comment:
I'd suggest doing `set_to_job` unconditionally, thus:

```python
resource.set_to_job(j)
if (memory_override := …)
    assert isinstance(memory_override, int)
    # Hail will select the right number of CPUs based on RAM request
    j.memory(f'{memory_override}G')
```

This ensures that in the (unlikely) event more unrelated settings are added to `resource` in future, they don't get ignored when the override is set.

(Unless it's important that `j.cpu(…)` is not called so that `j._cpu` stays None in order for that “Hail will select” to happen, in which case it might be good to mention that in the comment.)

Reply:
I've added a clarifying docstring here - AFAIK by setting either CPU or memory Hail will find a machine configuration to satisfy that based on the fixed CPU-RAM ratio, so no need to set both

Reply:
I had a second look and yeah, makes sense to set everything, then override if needed. I'm not sure which other options are going to appear in `resource`, but just in case